### PR TITLE
Fix escaping of string based filters in LoupeSearcher

### PR DESCRIPTION
**Pull request by alexander-schranz**
see #292 and #293

### CODE REVIEW

File: packages/seal-loupe-adapter/src/LoupeSearcher.php [hunk 1]
```diff
@@ -122,6 +122,15 @@ public function search(Search $search): Result
         );
     }
 
+    private function escapeFilterValue(string|int|float|bool $value): string
+    {
+        return match (true) {
+            \is_string($value) => '"' . \addslashes($value) . '"',
```
Comment (alexander-schranz) [inline packages/seal-loupe-adapter/src/LoupeSearcher.php]: `addslashes` sadly not enough still run into:

```
Loupe\Loupe\Exception\FilterFormatException: Col 7: Error: Expected valid string or float value, got '"'
```

The test uses a very special string:

```php
$value = "The 17\" O'Conner && O`Series \n OR a || 1%2 book?";
```

@Toflar do you have a hint for me? PS: Loupe 0.4

```bash
object(Loupe\Loupe\SearchParameters)#524 (10) {
  ["attributesToHighlight":"Loupe\Loupe\SearchParameters":private]=>
  array(0) {
  }
  ["attributesToRetrieve":"Loupe\Loupe\SearchParameters":private]=>
  array(1) {
    [0]=>
    string(1) "*"
  }
  ["attributesToSearchOn":"Loupe\Loupe\SearchParameters":private]=>
  array(1) {
    [0]=>
    string(1) "*"
  }
  ["filter":"Loupe\Loupe\SearchParameters":private]=>
  string(59) "tags = "The 17\" O\'Conner && O`Series
 OR a || 1%2 book?""
  ["hitsPerPage":"Loupe\Loupe\SearchParameters":private]=>
  int(20)
  ["page":"Loupe\Loupe\SearchParameters":private]=>
  int(1)
  ["query":"Loupe\Loupe\SearchParameters":private]=>
  string(0) ""
  ["showMatchesPosition":"Loupe\Loupe\SearchParameters":private]=>
  bool(false)
  ["showRankingScore":"Loupe\Loupe\SearchParameters":private]=>
  bool(false)
  ["sort":"Loupe\Loupe\SearchParameters":private]=>
  array(1) {
    [0]=>
    string(15) "_relevance:desc"
  }
}
```

Comment (Toflar) [reply packages/seal-loupe-adapter/src/LoupeSearcher.php]: I think Loupe should provide escaping itself for the filter. I'll check if I can implement this.

Comment (alexander-schranz) [reply packages/seal-loupe-adapter/src/LoupeSearcher.php]: If I use '"'  around the string and still escaping the `"` in `17"` to `17\"` does not have effect and fails with:

>  Loupe\Loupe\Exception\FilterFormatException: Col 7: Error: Expected valid string or float value, got '"'

If using `'` it fails in `'O'Conner'` even it is escaped to `'O\'Conner'`:

> Loupe\Loupe\Exception\FilterFormatException: Col 20: Error: Expected filterable attribute, got 'Conner'

@toflar a provided escaping function would be great.

Comment (Toflar) [reply packages/seal-loupe-adapter/src/LoupeSearcher.php]: Yes, the parser is not able to do this atm.

Comment (alexander-schranz) [reply packages/seal-loupe-adapter/src/LoupeSearcher.php]: I see, keep me uptodate :)

Comment (Toflar) [reply packages/seal-loupe-adapter/src/LoupeSearcher.php]: Can you test with https://github.com/loupe-php/loupe/pull/54? :)

**Pull request merged**